lib/test-setup.js

```diff
@@ -39,54 +40,53 @@ global.constants = {
 
 const defaultConstantsModule = Object.assign({}, require("./constants"), {
   RATE_PERIOD: 500,
+  RATE_LIMIT: 2,
+  RATE_WAIT: 10,
   MIN_HEARTBEAT_PERIOD: 0
 });
 global.constantsModule = Object.assign({}, defaultConstantsModule);
 mockRequire("./constants", global.constantsModule);
 
-global.mocks = {
-  deleteMessage: (AWS.SQS.prototype.deleteMessage = sinon.stub()),
-  getItem: (AWS.DynamoDB.DocumentClient.prototype.get = sinon.stub()),
-  putItem: (AWS.DynamoDB.DocumentClient.prototype.put = sinon.stub()),
-  deleteItem: (AWS.DynamoDB.DocumentClient.prototype.delete = sinon.stub()),
-  getQueueAttributes: (AWS.SQS.prototype.getQueueAttributes = sinon.stub()),
-  getQueueUrl: (AWS.SQS.prototype.getQueueUrl = sinon.stub()),
-  getSignedUrl: (AWS.S3.prototype.getSignedUrl = sinon.stub()),
-  putObject: (AWS.S3.prototype.putObject = sinon.stub()),
-  deleteObject: (AWS.S3.prototype.deleteObject = sinon.stub()),
-  requestPost: (request.post = sinon.stub()),
-  sendMessage: (AWS.SQS.prototype.sendMessage = sinon.stub()),
-  receiveMessage: (AWS.SQS.prototype.receiveMessage = sinon.stub()),
-  invoke: (AWS.Lambda.prototype.invoke = sinon.stub())
-};
-
+global.mocks = {};
 global.makePromiseFn = out => ({ promise: () => Promise.resolve(out) });
+global.makePromiseStub = out => sinon.stub().returns(makePromiseFn(out));
 
 global.resetMocks = () => {
   const {
     mocks,
     makePromiseFn,
+    makePromiseStub,
     constants: { QueueUrl, QueueAttributes, MessageId, ETag }
   } = global;
 
   Object.assign(global.constantsModule, defaultConstantsModule);
   Object.assign(process.env, global.env);
   Object.values(global.mocks).forEach(mock => mock.resetHistory());
 
-  mocks.requestPost.resolves({});
+  const pSQS = AWS.SQS.prototype;
+  const pS3 = AWS.S3.prototype;
+  const pDocumentClient = AWS.DynamoDB.DocumentClient.prototype;
+  const pLambda = AWS.Lambda.prototype;
 
-  mocks.deleteMessage.returns(makePromiseFn({}));
-  mocks.deleteItem.returns(makePromiseFn({}));
-  mocks.putItem.returns(makePromiseFn({}));
-  mocks.getItem.returns(makePromiseFn({}));
-  mocks.getQueueAttributes.returns(
-    makePromiseFn({ Attributes: QueueAttributes })
-  );
-  mocks.getQueueUrl.returns(makePromiseFn({ QueueUrl }));
-  mocks.getSignedUrl.returns("");
-  mocks.putObject.returns(makePromiseFn({ ETag }));
-  mocks.deleteObject.returns(makePromiseFn({}));
-  mocks.sendMessage.returns(makePromiseFn({ MessageId }));
-  mocks.receiveMessage.returns(makePromiseFn({ MessageId }));
-  mocks.invoke.returns(makePromiseFn({}));
+  Object.assign(mocks, {
```

Comment:
Shifted some creation of mocks around, because it appears that they were not actually getting reset before every test.